shared/actions/fs/index.js

```diff
@@ -690,96 +691,108 @@ const closeDestinationPicker = (state, action) => {
-  action.payload.routePath
-    ? RouteTreeGen.createPutActionIfOnPath({
-        expectedPath: action.payload.routePath,
-        otherAction: RouteTreeGen.createNavigateAppend({path: ['sendAttachmentToChat']}),
-      })
-    : RouteTreeGen.createNavigateAppend({path: ['sendAttachmentToChat']})
-
-function* showSendLinkToChat(state, action) {
+const initSendLinkToChat = (state, action) => {
   const elems = Types.getPathElements(state.fs.sendLinkToChat.path)
-  const routeChange = Saga.put(
-    action.payload.routePath
-      ? RouteTreeGen.createPutActionIfOnPath({
-          expectedPath: action.payload.routePath,
-          otherAction: RouteTreeGen.createNavigateAppend({path: ['sendLinkToChat']}),
-        })
-      : RouteTreeGen.createNavigateAppend({path: ['sendLinkToChat']})
-  )
   if (elems.length < 3 || elems[1] === 'public') {
-    // Not a TLF, or a public TLF; just show the modal and let user copy the path.
-    yield routeChange
+    // Not a TLF, or a public TLF; just let user copy the path.
     return
   }
 
-  const actions = [routeChange]
-
   if (elems[1] !== 'team') {
     // It's an impl team conversation. So resolve to a convID directly.
-    actions.push(
-      Saga.callUntyped(function*() {
-        const result = yield Saga.callPromise(RPCChatTypes.localFindConversationsLocalRpcPromise, {
-          identifyBehavior: RPCTypes.tlfKeysTLFIdentifyBehavior.chatGui,
-          membersType: RPCChatTypes.commonConversationMembersType.impteamnative,
-          oneChatPerTLF: false,
-          tlfName: elems[2],
-          topicName: '',
-          topicType: RPCChatTypes.commonTopicType.chat,
-          visibility: RPCTypes.commonTLFVisibility.private,
-        })
-
-        if (!result.conversations || !result.conversations.length) {
-          // TODO: error?
-          return
-        }
-
-        yield Saga.put(
-          FsGen.createSetSendLinkToChatConvID({
-            convID: ChatTypes.conversationIDToKey(result.conversations[0].info.id),
-          })
-        )
+    return RPCChatTypes.localFindConversationsLocalRpcPromise({
+      identifyBehavior: RPCTypes.tlfKeysTLFIdentifyBehavior.chatGui,
+      membersType: RPCChatTypes.commonConversationMembersType.impteamnative,
+      oneChatPerTLF: false,
+      tlfName: elems[2],
+      topicName: '',
+      topicType: RPCChatTypes.commonTopicType.chat,
+      visibility: RPCTypes.commonTLFVisibility.private,
+    }).then(result =>
+      // This action, no matter setting a real idKey or
+      // noConversationIDKey, causes a transition into 'read-to-send'
+      // state, which is what we want here. If we don't have a
+      // conversation we should create it when user tries to send.
+      FsGen.createSetSendLinkToChatConvID({
+        convID:
+          result.conversations && result.conversations.length
+            ? ChatTypes.conversationIDToKey(result.conversations[0].info.id)
+            : ChatConstants.noConversationIDKey,
       })
     )
```

Comment:
Should we also catch an error so that we can process and display it?

Reply:
Good question; note that chat doesn't catch the error here: https://github.com/keybase/client/blob/d0a8481657b695448c68125250c4173a3bdf0654/shared/actions/chat2/index.js#L1586

So I'm guessing either this RPC is not expect to fail, i.e. if it fails we'd want a black bar, or somewhere else we're filtering on this error to not show a blackbar.

Reply:
I know but it seems like an opportunity to reduce the number of black bars.

Reply:
You mean if it's not expected to fail? Catching it would just show a red bar instead of black bar. I feel like in that case we should just blackbar and fix it if needed.